Miracast/MiracastService/MiracastService.cpp

```diff
@@ -0,0 +1,1043 @@
+						}
+					}
+					m_miracast_ctrler_obj->m_ePlayer_state = MIRACAST_PLAYER_STATE_STOPPED;
+					m_eService_state = MIRACAST_SERVICE_STATE_DISCOVERABLE;
+				}
+				else if (player_state == "INITIATED" || player_state == "initiated")
+				{
+					m_miracast_ctrler_obj->m_ePlayer_state = MIRACAST_PLAYER_STATE_INITIATED;
+				}
+				else if (player_state == "INPROGRESS" || player_state == "inprogress")
+				{
+					m_miracast_ctrler_obj->m_ePlayer_state = MIRACAST_PLAYER_STATE_INPROGRESS;
+				}
+				else
+				{
+					m_miracast_ctrler_obj->m_ePlayer_state = MIRACAST_PLAYER_STATE_IDLE;
+				}
+			}
+
+			if ( m_isServiceEnabled && restart_discovery_needed )
+			{
+				// It will restart the discovering
+				m_miracast_ctrler_obj->restart_session_discovery();
+			}
+
+			MIRACASTLOG_INFO("Player State set to [%s (%d)] for Source device [%s].", player_state.c_str(), (int)m_miracast_ctrler_obj->m_ePlayer_state, mac.c_str());
+			// @TODO: Need to check what to do next?
+
+			MIRACASTLOG_INFO("Exiting..!!!");
+			returnResponse(success);
+		}
+
+		uint32_t MiracastService::setLogging(const JsonObject &parameters, JsonObject &response)
+		{
+			std::string log_level = "";
+			bool success = false;
+
+			MIRACASTLOG_INFO("Entering..!!!");
+
+			if (parameters.HasLabel("separate_logger"))
+			{
+				JsonObject separate_logger;
+				separate_logger = parameters["separate_logger"].Object();
+
+				if (separate_logger.HasLabel("status"))
+				{
+					std::string status = "";
+					status = separate_logger["separate_logger"].String();
+
+					success = true;
+
+					if (status == "ENABLE" || status == "enable")
+					{
+						std::string logfilename = "";
+						logfilename = separate_logger["logfilename"].String();
+						MIRACAST::enable_separate_logger(logfilename);
+					}
+					else if (status == "DISABLE" || status == "disable")
+					{
+						MIRACAST::disable_separate_logger();
+					}
+					else
+					{
+						success = false;
+					}
+				}
+			}
+
+			if (parameters.HasLabel("level"))
+			{
+				LogLevel level = FATAL_LEVEL;
+				getStringParameter("level", log_level);
+				success = true;
+				if (log_level == "FATAL" || log_level == "fatal")
+				{
+					level = FATAL_LEVEL;
+				}
+				else if (log_level == "ERROR" || log_level == "error")
+				{
+					level = ERROR_LEVEL;
+				}
+				else if (log_level == "WARNING" || log_level == "warning")
+				{
+					level = WARNING_LEVEL;
+				}
+				else if (log_level == "INFO" || log_level == "info")
+				{
+					level = INFO_LEVEL;
+				}
+				else if (log_level == "VERBOSE" || log_level == "verbose")
+				{
+					level = VERBOSE_LEVEL;
+				}
+				else if (log_level == "TRACE" || log_level == "trace")
+				{
+					level = TRACE_LEVEL;
+				}
+				else
+				{
+					success = false;
+				}
+
+				if (success)
+				{
+					set_loglevel(level);
+				}
+			}
+
+			MIRACASTLOG_INFO("Exiting..!!!");
+			returnResponse(success);
+		}
+
+#ifdef ENABLE_MIRACAST_SERVICE_TEST_NOTIFIER
```

Comment:
are we enabling this flag for release build?

Reply:
Currently It is enabled. Now disabled it as part of https://github.com/rdkcentral/rdkservices/pull/4618/commits/56e11f35b7a6a7ea1fcc73da62a69f46ef3663d5 and https://github.com/rdkcentral/rdkservices/pull/4618/commits/af878fc3ccc6c61fff851af1f44dde68b5b892e3